djvm/src/main/kotlin/net/corda/djvm/source/SourceClassLoader.kt

```diff
@@ -120,18 +110,16 @@ class SourceClassLoader private constructor(
                 logger.error("Denying request for actual {}", name)
                 return null
             }
+        } else if (isJvmInternal(name)) {
+            /**
+             * Without a special [BootstrapClassLoader], we need
+             * to fetch Java API classes from the JVM itself.
+             */
+            return getSystemClassLoader().getResource(name)
         }
 
         return parent?.getResource(name) ?: findResource(name)
     }
-
-    /**
-     * Deny all requests for DJVM classes from any user-supplied jars.
-     */
-    override fun findResource(name: String): URL? {
-        return if (name.startsWith("net/corda/djvm/")) null else super.findResource(name)
-    }
-
 }
 
 private fun resolvePaths(paths: List<Path>): Array<URL> {
```

Comment:
Potentially the PR could consist just of the changes in this file and test, correct?

Reply:
The main changes are to `SourceClassLoader` _and_ `SandboxClassLoader`. `TestBase` can now be initialised for `JAVA` or `KOTLIN`, which determines what is loaded into `SourceClassLoader`. And this obviously has a knock-on effect for most of the tests.